yann/utils/raster.py

```diff
@@ -98,7 +98,7 @@ def tile_raster_images(X, img_shape, tile_shape, tile_spacing=(0, 0),
                                     dtype='uint8')
         else:
             out_array = numpy.zeros((out_shape[0], out_shape[1], 4),
-                                    dtype=X.dtype)
+                                    dtype=X[1].dtype)
```

Comment:
Yeah, If you check a couple of lines before that you are checking if X is a tuple "if isinstance(X, tuple)". This code will be executed only if X is a tuple, when it does it's throwing an error that tuple don't have a dtype. That's why I used that X[1] for dtype.